crypto/bls_test.go

```diff
@@ -299,3 +299,171 @@ func TestRemovePubKeys(t *testing.T) {
 		fmt.Sprintf("incorrect key %s, should be %s, keys are %s",
 			partialPk, BLSkey, pks))
 }
+
+// BLS multi-signature
+// batch verification
+//
+// Verify n signatures of the same message under different keys using the fast
+// batch verification technique and compares the result to verifying each signature
+// separately.
+func TestBatchVerify(t *testing.T) {
+	mrand.Seed(time.Now().UnixNano())
+	// random message
+	input := make([]byte, 100)
+	_, err := mrand.Read(input)
+	require.NoError(t, err)
+	// hasher
+	kmac := NewBLSKMAC("test tag")
+	// number of signatures to aggregate
+	sigsNum := mrand.Intn(100) + 1
```

Comment:
should we set the minimum higher than 1, if we're testing multi/batch? even though getting 1 is unlikely.

Reply:
`0` and `1` are edge cases that should be supported by the function.
I added a specific test for `0` but I included `1` in the generic case. The logic will still be the same with `1` signature (building a tree etc.)